src/test/java/suite/operator/TestData.java

```diff
@@ -27,6 +39,46 @@ public static Set<Arguments> baseArgs() {
         return args;
     }
 
+    public static Stream<Arguments> sharingArgs() {
+        final var privateOperator = named("PRIVATE OPERATOR", getOperator(Operator.Type.PRIVATE).orElse(null));
+        final var publicOperator = named("PUBLIC OPERATOR", getOperator(Operator.Type.PUBLIC).orElse(null));
+
+        final var client = named("CLIENT", CLIENT);
+        final var sharingRecipient = named("SHARING RECIPIENT", SHARING_RECIPIENT);
+        final var nonSharingRecipient = named("NON-SHARING RECIPIENT", NON_SHARING_RECIPIENT);
+
+        return Stream.of(
+                // CLIENT shares with CLIENT.
+                Arguments.of(client, privateOperator, client, privateOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, privateOperator, client, publicOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, publicOperator, client, privateOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, publicOperator, client, publicOperator, DecryptionStatus.SUCCESS),
+
+                // CLIENT shares with SHARING RECIPIENT.
+                Arguments.of(client, privateOperator, sharingRecipient, publicOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, publicOperator, sharingRecipient, publicOperator, DecryptionStatus.SUCCESS),
+
+                // CLIENT does not share with NON-SHARING RECIPIENT.
```

Comment:
are some public/private operator combinations intentionally skipped? should we add a comment about that?

Reply:
Yes, added comments to explain why